unit/deserialize.go

```diff
@@ -252,7 +252,7 @@ func (l *lexer) lexNextSectionOrOptionFunc(section string) lexStep {
 			return l.ignoreLineFunc(l.lexNextSectionOrOptionFunc(section)), nil
 		}
 
-		l.buf.UnreadRune()
+		_ = l.buf.UnreadRune()
```

Comment:
might be worth to a comment here saying that this should never error as we called ReadRune() before